File: android/src/main/kotlin/com/cleancode/liveness_azure_flutter/LivenessResultModel.kt
package com.cleancode.liveness_azure_flutter

import android.os.Parcelable
import com.azure.android.ai.vision.faceanalyzer.LivenessFailureReason
import com.azure.android.ai.vision.faceanalyzer.LivenessStatus
import com.azure.android.ai.vision.faceanalyzer.RecognitionStatus
import kotlinx.parcelize.Parcelize

@Parcelize
data class LivenessResultModel(
    val livenessStatus: LivenessStatus,
    val failureReason: LivenessFailureReason,
    val verificationStatus: RecognitionStatus,
    val confidence: Float,
    val resultId: String,
    val digest: String,
    val faceUID: String,
) : Parcelable

File: ios/Classes/Model/LivenessResultModel.swift
class LivenessResultModel: Codable{
    var livenessStatus: String?
    var failureReason: String?
    var verificationStatus: String?
    var confidence: Float?
    var resultId: String?
    var digest: String?
    var faceUID: String?
    
    init(livenessStatus: String?,
         failureReason: String?,
         verificationStatus: String?,
         confidence: Float?,
         resultId: String?,
         digest: String?,
         faceUID: String?) {
        
        self.livenessStatus = livenessStatus
        self.failureReason = failureReason
        self.verificationStatus = verificationStatus
        self.confidence = confidence
        self.resultId = resultId
        self.digest = digest
        self.faceUID = faceUID
    }
}
